### ISSUE TITLE
Reference doc needs some examples

### ISSUE THREAD

**Issue by sl0m0ZA** [cirrax/puppet-dovecot#3]
I felt that the existing reference was hard to follow with regards to "sections" and filename allocation was not very clear either. 
I have submitted a pull request with added documentation in the REFERENCE.md file.
I hope this proposal adds value to your project. 
Thank you for your efforts.

**Comment by sl0m0ZA** [comment]
I have moved the documentation to the relevant .pp files for the autogeneration of REFERENCE.md by jdk

**Comment by sl0m0ZA** [comment]
Apologies - I could not find a way to group all 3 file changes in a single pull request @trefzer

**Comment by sl0m0ZA** [comment]
@trefzer Changes made as per requests

**Comment by trefzer** [comment]
merged #8 #7 #6, so close

**Comment by sl0m0ZA** [comment]
Thank you
